sql/core/src/main/scala/org/apache/spark/sql/parquet/CatalystSchemaConverter.scala

```diff
@@ -446,7 +446,7 @@ private[parquet] class CatalystSchemaConverter(
           field.name,
           Types
             .buildGroup(REPEATED)
-            .addField(convertField(StructField("element", elementType, nullable)))
+            .addField(convertField(StructField("array", elementType, nullable)))
```

Comment:
Actually I made a mistake here. We should leave this `array` as `array_element`.

This is a little bit complicated... So in the early days, when Spark SQL Parquet support was firstly authored, Parquet format spec wasn't clear about how to write arrays and maps. So Spark SQL took a somewhat weird approach here: if the array may contain nulls, we mimic parquet-hive, which writes a 3-level structure with `array_element` as the 2nd level type name; otherwise, we mimic parquet-avro, which writes a 2-level structure with `array` as the 2nd level type name.